pkg/river/types.go

```diff
@@ -26,6 +28,22 @@ type Unmarshaler interface {
 	UnmarshalRiver(f func(v interface{}) error) error
 }
 
+// The Defaulter interface allows a type to implement default functionality
+// in River evaluation.
```

Comment:
Q1: If my arguments have a block, then a `SetToDefault()` will be called for it and all its nested blocks?

Q2: If my arguments have a `squash`-ed block will `SetToDefault()` be called on that squashed struct and its nested blocks?

Would be good to document answers to these.

Reply:
added documentation